lib/tasks/admin.rake

```diff
@@ -107,7 +107,15 @@ namespace :admin do
     # Create jurisdiction for it does not already exist
     if jurisdiction == nil
       jurisdiction = Jurisdiction.create(name: jur_name , parent: parent)
-      unique_identifier = Digest::SHA256.hexdigest(jurisdiction.jurisdiction_path_string)
+
+      # create a 10 character, url-safe, base-64 string based on the SHA-256 hash of the jurisdiction path
+      unique_identifier = Base64::urlsafe_encode64([[Digest::SHA256.hexdigest(jurisdiction.jurisdiction_path_string)].pack('H*')].pack('m0'))[0, 10]
+
+      # Warn user if collision has occured
+      if Jurisdiction.where('BINARY unique_identifier = ?', unique_identifier).where.not(id: jurisdiction.id).any?
+        puts "JURISDICTION IDENTIFIER HASH COLLISION FOR: #{jurisdiction[:path]}"
```

Comment:
In this case, should we return? As it is now, we still update the jurisdiction with the new (but duplicate) identifier.

Reply:
I think so. @mayerm94 can you confirm?

Reply:
I think we just need to make sure that if it returns, it wouldn't have imported some jurisdictions but not others, leaving us in a difficult state to recover from. All of this is wrapped in a transaction, if we just return though, will the changes before the return still go through? @holmesie

Reply:
If there was a problem in the transaction, raise an exception (don't return). That will prevent the whole transaction from being applied.

Reply:
Addressed here: https://github.com/SaraAlert/SaraAlert/pull/446/commits/38dc5f58019d8aacda9df4952d3b9152de1997fb